lib/models/blueprint.js

```diff
@@ -430,12 +443,82 @@ let Blueprint = CoreObject.extend({
     let fileInfos = await process.call(this, intoDir, locals);
 
     // commit changes for each FileInfo (with prompting as needed)
-    await Promise.all(fileInfos.map((fi) => this._commit(fi)));
+    await Promise.all(fileInfos.map((info) => this._commit(info)));
 
     // run afterInstall/afterUninstall userland hooks
     await afterHook.call(this, options);
   },
 
+  /**
+    @private
+    @method shouldConvertToJS
+    @param {Object} options
+    @param {FileInfo} fileInfo
+    @return {Boolean}
+  */
+  shouldConvertToJS(options, fileInfo) {
+    // If this isn't turned on, it doesn't matter what else was passed, we're not touching it.
+    if (!this.shouldTransformTypeScript) {
+      return false;
+    }
+
+    // Since the `--typescript` is Boolean, ember-cli automatically generates a `--no-typescript` flag
+    // as well. This means that `options.typescript` will be undefined if *nothing* is passed, but
+    // will be explicitly false if `--no-typescript` is passed. We use this variable to differentiate
+    // between both cases
+    const noTypeScript = options.typescript === false;
+
+    // If the user wants TypeScript output but there is no TypeScript blueprint available, we want
+    // to warn them that they're not going to get what they're expecting while still at least giving
+    // them the JS output. We check for this *after* checking `shouldTranformTypeScript` because
+    // it's possible for people to set `{typescript: true}` in their `.ember-cli` file, which would
+    // the erroneously trigger this message every time they generate a JS blueprint even though
+    // they didn't pass the flag.
+    if (!noTypeScript && !isTypeScriptFile(fileInfo.inputPath)) {
+      this.ui.writeLine(
+        chalk.yellow(
+          "You passed the '--typescript' flag but there is no TypeScript blueprint available. " +
+            'A JavaScript blueprint will be generated instead.'
+        )
+      );
+      return false;
+    }
+
+    // Indicates when the user has asked for TypeScript either globally (by setting
+    // `isTypeScriptProject` to true) or locally (by passing the `--typescript` flag when they
+    // invoked the generator). Since ember-cli merges `.ember-cli` and the all of the flag values into
+    // one object, we could theoretically *just* have a `typescript` flag that is either passed in
+    // to `generate` OR set in `.ember-cli`, but we thought the DX would be improved by differentiating
+    // between what is intended to be global vs. local config. Note that, because of this behavior,
+    // the user could *technically* also just set `typescript: true` in `.ember-cli` or pass
+    // `--is-typescript-project` when invoking `generate`, but we don't expect this to happen often.
```

Comment:
Can you rework this comment slightly? Specifically removing the alternative that we discarded, and just focusing on the goal / intent of the actual code we are going with...